internal/command/user_grant.go

```diff
@@ -77,32 +86,56 @@ func (c *Commands) ChangeUserGrant(ctx context.Context, userGrant *domain.UserGr
 	return userGrantWriteModelToUserGrant(changedUserGrant), nil
 }
 
-func (c *Commands) changeUserGrant(ctx context.Context, userGrant *domain.UserGrant, resourceOwner string, cascade bool) (_ eventstore.Command, _ *UserGrantWriteModel, err error) {
+func (c *Commands) changeUserGrant(ctx context.Context, userGrant *domain.UserGrant, resourceOwner *string, cascade, ignoreUnchanged bool, check UserGrantPermissionCheck) (_ eventstore.Command, _ *UserGrantWriteModel, err error) {
 	if userGrant.AggregateID == "" {
 		return nil, nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-3M0sd", "Errors.UserGrant.Invalid")
 	}
-	existingUserGrant, err := c.userGrantWriteModelByID(ctx, userGrant.AggregateID, userGrant.ResourceOwner)
+	existingUserGrant, err := c.userGrantWriteModelByID(ctx, userGrant.AggregateID, gu.Value(resourceOwner))
 	if err != nil {
 		return nil, nil, err
 	}
-	err = checkExplicitProjectPermission(ctx, existingUserGrant.ProjectGrantID, existingUserGrant.ProjectID)
-	if err != nil {
-		return nil, nil, err
+	if check == nil {
```

Comment:
Why do we separate the permission check? We can just do the permission check once at the point where we have all necessary information?

Reply:
This is about not breaking deprecated methods

Reply:
You anyway do a permission check, so how is this different?

Reply:
the permission check is done before a not found or a precondition error is returned. I didn't change this.

Reply:
like this? https://github.com/zitadel/zitadel/pull/9978/commits/f71485c3274b3303a6cd14d11fc7989f76b6adf2